pkg/proxy/iptables/proxier.go

```diff
@@ -992,9 +997,9 @@ func (proxier *Proxier) syncProxyRules() {
 			// If the "external" IP happens to be an IP that is local to this
 			// machine, hold the local port open so no other process can open it
 			// (because the socket might open but it would never work).
-			if local, err := utilproxy.IsLocalIP(externalIP); err != nil {
-				klog.Errorf("can't determine if IP is local, assuming not: %v", err)
-			} else if local && (svcInfo.Protocol() != v1.ProtocolSCTP) {
+			if len(localAddrs) == 0 {
+				klog.Errorf("couldn't find any local IPs, assuming external IP %s is not local", externalIP)
```

Comment:
You already logged an error about `GetLocalAddrs` failing above. Do we need to log an error for each external IP too? (Maybe just add "assuming all external IPs are not local" to the earlier message.)
(Likewise in ipvs.)

Reply:
Good catch, thanks